### DISCUSSION TITLE
How to remove a specific path ?

### DISCUSSION THREAD

**Discussion by bchateli** [NVlabs/sionna#382] [Q&A] [answered]
Hi everyone,

When using sionna.rt, is there a simple way to remove a path from the path object obtained using scene.compute_paths() ?

Typically, with one Transmitter and one Receiver on the same elevation plane, one could want to keep only the paths on this elevation plane.

Thank you in advance.

**Comment by jhoydis** [comment, marked as answer]
Hi @bchateli,

The easiest way to do this would be to set the channel gain corresponding to this path to zero.

Do you need this for visualisation or evaluation purposes?

**Reply by bchateli** [reply]
Hi @jhoydis, thank you for the quick answer. I need this for both. For evaluation purposes, I agree with your answer.

For visualisation, in a simple scenario (mono-antenna Tx/Rx), I came up with the following:
```python
paths.mask = tf.gather(paths.mask,idx_keep,axis=-1)
paths.types = tf.gather(paths.types,idx_keep,axis=-1)
paths.a = tf.gather(paths.a,idx_keep,axis=-2)
paths.tau = tf.gather(paths.tau,idx_keep,axis=-1)
paths.theta_t = tf.gather(paths.theta_t,idx_keep,axis=-1)
paths.phi_t = tf.gather(paths.phi_t,idx_keep,axis=-1)
paths.theta_r = tf.gather(paths.theta_r,idx_keep,axis=-1)
paths.phi_r = tf.gather(paths.phi_r,idx_keep,axis=-1)
paths.vertices = tf.gather(paths.vertices,idx_keep,axis=-2)
paths.objects = tf.gather(paths.objects,idx_keep,axis=-1)
```
where idx_keep is a selection vector for the wanted paths

**Reply by jhoydis** [reply]
Great if that works for you.

I think that you probably do not need to set all of these properties to remove a path from the visualisation. Updating the `mask` might be sufficient. I would need to look into the code of the preview function.
